packages/editor/src/internal-utils/parse-blocks.ts

```diff
@@ -60,21 +60,45 @@ function parseBlockObject({
     return undefined
   }
 
-  if (
-    blockObject._type === context.schema.block.name ||
-    blockObject._type === 'block' ||
-    !context.schema.blockObjects.some(({name}) => name === blockObject._type)
-  ) {
+  const schemaType = context.schema.blockObjects.find(
+    ({name}) => name === blockObject._type,
+  )
+
+  if (!schemaType) {
     return undefined
   }
 
+  // Validates all props on the block object and only takes those that match
+  // the name of a field
+  const values = schemaType.fields.reduce<Record<string, unknown>>(
+    (fieldValues, field) => {
+      fieldValues[field.name] = blockObject[field.name]
+      return fieldValues
+    },
+    {},
+  )
+
+  // Takes all hidden props (props that start with _) on the block object
+  const hiddenValues = Object.keys(blockObject).reduce<Record<string, unknown>>(
+    (hiddenValues, key) => {
+      if (key.startsWith('_') && key !== '_type' && key !== '_key') {
+        hiddenValues[key] = blockObject[key]
+      }
+
+      return hiddenValues
+    },
+    {},
+  )
+
   return {
-    ...blockObject,
+    _type: schemaType.name,
     _key: options.refreshKeys
       ? context.keyGenerator()
       : typeof blockObject._key === 'string'
         ? blockObject._key
         : context.keyGenerator(),
+    ...values,
+    ...hiddenValues,
```

Comment:
Wont you still have the canvas cp bug here? Since you retain _map ect?

Reply:
Yes. This PR retains all props starting with `_`.

Reply:
I've updated the PR now so all fields have to be defined, even those for props starting with `_`.